pkg/apis/core/types.go

```diff
@@ -486,6 +486,24 @@ const (
 	PersistentVolumeClaimFileSystemResizePending PersistentVolumeClaimConditionType = "FileSystemResizePending"
 )
 
+type PersistentVolumeClaimResizeStatus string
+
+const (
+	// When expansion is complete, the empty string is set by resize controller or kubelet.
+	PersistentVolumeClaimNoExpansionInProgress PersistentVolumeClaimResizeStatus = ""
+	// State set when resize controller starts expanding the volune in control-plane
+	PersistentVolumeClaimControllerExpansionInProgress PersistentVolumeClaimResizeStatus = "ControllerExpansionInProgress"
+	// State set when expansion has failed in resize controller with a terminal error.
+	// Transient errors such as timeout don't set ControllerExpansionFailed.
+	PersistentVolumeClaimControllerExpansionFailed PersistentVolumeClaimResizeStatus = "ControllerExpansionFailed"
+	// State set when resize controller has finished expanding the volume but further expansion is needed on the node.
```

Comment:
maybe mention kubelet has not yet started node expansion to differentiate from NodeExpansionInProgress?

Reply:
I think that current comment is accurate, in the sense that resize controller in control-plane is finished with the work but mentioning the fact that "kubelet has not yet started node expansion" borders into an implementation detail which is subject to change and I am not comfortable putting it in API docs. What do you think?